rust/arrow/src/compute/kernels/arithmetic.rs

```diff
@@ -147,16 +143,17 @@ where
         .values()
         .iter()
         .zip(right.values().iter())
-        .map(|(l, r)| op(*l, *r))
-        .collect::<Vec<T::Native>>();
+        .map(|(l, r)| op(*l, *r));
+    let mut buffer = MutableBuffer::new(0);
+    buffer.extend_from_exact_iter(values);
```

Comment:
This is the main difference in this PR for the arithmetics: extend from the iterator instead of extending a vector and then memcopying the data.